File: hw/chiriboga/u1/hw10SaveAnObjects/SaveAnObjectPython/ec/edu/espe/saveanobject/menu/Menu.py
from utils import SerializeUtil

def showMenu():
    while True:
        print("Menu:")
        print("1. Create Computer")
        print("2. Read Computer")
        print("3. Exit")

        try:
            operation = int(input("Enter the option: "))
        except ValueError:
            print("Invalid operation.")
            continue

        if operation == 1:
            SerializeUtil.create(input)
        elif operation == 2:
            SerializeUtil.read(input)
        elif operation == 3:
            print("Exiting...")
            break
        else:
            print("Invalid operation.")

if __name__ == "__main__":
    showMenu()

File: hw/chiriboga/u1/HomeworksUnit1/hw10SaveAnObjects/SaveAnObject/src/ec/edu/espe/saveanobject/menu/Menu.java
package ec.edu.espe.saveanobject.menu;

import java.util.Scanner;
import utils.SerializeUtil;

/**
 * Menu class to handle user interaction
 */
public class Menu {
    public static void showMenu() {
        try (Scanner scanner = new Scanner(System.in)) {
            int operation;

            do {
                System.out.println("Menu:");
                System.out.println("1. Create Computer");
                System.out.println("2. Read Computer");
                System.out.println("3. Exit");

                System.out.print("Enter the option: ");
                operation = scanner.nextInt();
                scanner.nextLine();

                switch (operation) {
                    case 1:
                        SerializeUtil.create(scanner);
                        break;
                    case 2:
                        SerializeUtil.read(scanner);
                        break;
                    case 3:
                        System.out.println("Exiting...");
                        break;
                    default:
                        System.out.println("Invalid operation.");
                }
            } while (operation != 3);
        }
    }
}

